src/Deserializers/SnakDeserializer.php

```diff
@@ -114,12 +108,24 @@ private function newValueSnak( array $serialization ) {
 		);
 	}
 
+	/**
+	 * @param array $serialization
+	 *
+	 * @return DataValue
+	 */
 	private function deserializeDataValue( $serialization ) {
 		try {
 			return $this->dataValueDeserializer->deserialize( $serialization );
 		} catch ( DeserializationException $ex ) {
+			$value = isset( $serialization[DataValueDeserializer::VALUE_KEY] )
+				? $serialization[DataValueDeserializer::VALUE_KEY]
+				: null;
+			$type = isset( $serialization[DataValueDeserializer::TYPE_KEY] )
+				? $serialization[DataValueDeserializer::TYPE_KEY]
+				: null;
 			$error = isset( $serialization['error'] ) ? $serialization['error'] : $ex->getMessage();
-			return new UnDeserializableValue( $serialization['value'], $serialization['type'], $error );
+
+			return new UnDeserializableValue( $value, $type, $error );
```

Comment:
Even though the documentation for `UnDeserializableValue` is horrible (phpdoc not at all in line with the actual code), this seems valid.